include/bluetooth/bluetooth.h

```diff
@@ -518,6 +730,20 @@ struct bt_le_scan_cb {
  */
 int bt_le_scan_start(const struct bt_le_scan_param *param, bt_le_scan_cb_t cb);
 
+/** @brief Set scan timeout
+ *
+ *  Set timeout for the active or passive scanner. Once this timeout is reached
+ *  the scanner will be automatically stopped and the application will be
+ *  notified by the scan timeout callback.
+ *  Timeout will be rounded up to nearest multiple of 10 ms. Maximum timeout is
+ *  655.35 seconds
+ *
+ *  @param timeout  Timeout in ms, K_FOREVER or K_NO_WAIT will disable timeout.
+ *
+ *  @return Zero on success or error code otherwise.
+ */
+int bt_le_set_scan_timeout(s32_t timeout);
```

Comment:
I wonder why this the timeout is not part of the bt_le_scan_param? Or you have consider it but though a dedicated one might be better? If we do this way the applications might need to read the timeout in case it persists, which I assume it does otherwise it would only be valid to use this API while scanning which is probably racy so most likely calls to this function should fail if there is already an scan ongoing so I don't see much of a benefit of not adding a timeout directly into bt_le_scan_param and handle it per instance.

Reply:
I think the reason for why I added it here is no longer present.
I wanted to use scan parameters for bt_create_conn_le in order to provide options for scanning as well there, but I ended up creating a different struct for conn_create scan parameters instead.

I can move scan timeout as a parameter back to bte_le_scan_param.

Any comments on the type of timeout. I switched to s32_t since it was a timeout parameter, but maybe I should go back to units and call it duration instead (this should be aligned with the timeout/duration parameter of the bt_le_adv_set_start function)?

Reply:
what about period? is this left unused on purpose?

Reply:
Currently left out on purpose. yes.